project_agile/models/project_task.py

```diff
@@ -0,0 +1,691 @@
+        relation='project_task_type2_sub_types_rel',
+        column1='type_id',
+        column2='sub_type_id',
+        string='Sub Types',
+    )
+
+    @api.model
+    def name_search(self, name='', args=None, operator='ilike', limit=100):
+        if args is None:
+            args = []
+
+        # The key 'selected_task_type_ids' is defined on the
+        # project.type form view.
+        if 'selected_task_type_ids' in self.env.context:
+            args.append((
+                'id', 'in', [
+                    x[1]
+                    for x in self.env.context['selected_task_type_ids'][0][2]
+                ]
+            ))
+
+        if 'board_project_ids' in self.env.context:
+            project_ids = self.env.context.get('board_project_ids', [])
+            if project_ids:
+                args.append([
+                    'id', 'in', self.env['project.project'].browse(
+                        project_ids[0][2]
+                    ).mapped('type_id.task_type_ids').fetch_all()
+                ])
+
+        return super(TaskType, self).name_search(
+            name=name, args=args, operator=operator, limit=limit
+        )
+
+    @api.multi
+    def fetch_all(self):
+        task_type_ids = []
+
+        def collect_task_types(tt):
+            if tt.id in task_type_ids:
+                return
+
+            task_type_ids.append(tt.id)
+            for subtype in tt.type_ids:
+                collect_task_types(subtype)
+
+        for task_type in self:
+            collect_task_types(task_type)
+
+        return task_type_ids
+
+
+class TaskPriority(models.Model):
+    _name = 'project.task.priority'
+    _description = 'Task Priority'
+    _inherit = ['project.agile.code_item']
+
+    type_ids = fields.Many2many(
+        comodel_name='project.task.type2',
+        relation='project_task_type2_task_priority_rel',
+        column1='priority_id',
+        column2='type_id',
+        string='Types',
+    )
+
+    icon = fields.Binary(
+        string='Icon'
+    )
+
+    _sql_constraints = [
+        ('project_task_priority_name_unique', 'unique(name)',
+         'Priority name already exists')
+    ]
+
+    @api.model
+    def name_search(self, name='', args=None, operator='ilike', limit=100):
+        if args is None:
+            args = []
+
+        # The key 'selected_priority_ids' is defined on the
+        # project.task.type2 form view.
+        if 'selected_priority_ids' in self.env.context:
+            args.append((
+                'id', 'in', self.env.context['selected_priority_ids'][0][2]
+            ))
+
+        return super(TaskPriority, self).name_search(
+            name=name, args=args, operator=operator, limit=limit
+        )
+
+
+class Task(models.Model):
+    _name = 'project.task'
+    _inherit = ['project.task', 'project.agile.mixin.id_search']
+
+    def _default_agile_order(self):
+        self.env.cr.execute("SELECT MAX(agile_order) + 1 FROM project_task")
```

Comment:
any reason to use SQL instead of standard ORM?

Reply:
I didn't want to fetch complete task just to read agile_order..

Reply:
Still: I would suggest if possible to use ORM functions

Reply:
Fixed